File: 1218/1218.cs
public class Solution {
    public int LongestArithmeticSubsequenceOf(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(LongestArithmeticSubsequenceOf(root.left), LongestArithmeticSubsequenceOf(root.right));
    }
}

File: 1218/1218.go
func longestArithmeticSubsequenceOf(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := longestArithmeticSubsequenceOf(root.Left)
	r := longestArithmeticSubsequenceOf(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}